packages/stylist/src/manager.rs

```diff
@@ -290,8 +361,59 @@ mod feat_ssr {
             Ok(())
         }
     }
+
+    impl StyleManagerBuilder {
+        /// Set the [StaticWriter] for current manager.
+        ///
+        /// # Note
+        ///
+        /// This also sets the StyleManager into the "static" mode. which it will stop rendering
+        /// styles into any html element.
+        pub fn writer(mut self, w: StaticWriter) -> Self {
+            self.writer = Some(w);
+
+            self
+        }
+    }
+
+    impl StyleManager {
+        /// Returns StyleData of current style manager.
+        ///
+        /// # Note
+        ///
+        /// If you are using [`ManagerProvider`](crate::yew::ManagerProvider),
+        /// this behaviour is managed automatically.
+        pub fn style_data(&self) -> StyleData {
+            self.inner.style_data()
+        }
+
+        /// Writes styles stored in the manager as `<style data-style="stylist-...">...</style>`.
+        pub fn write_static_markup<W>(&self, w: &mut W) -> fmt::Result
+        where
+            W: fmt::Write,
+        {
+            self.inner.write_static_markup(w)
+        }
+    }
+
+    impl Drop for StyleManagerBuilder {
```

Comment:
Writing the style data and markup on `Drop` seems dubious to me. There might be cases where the (shared) `StyleManagerBuilder` is alive beyond the rendering, e.g. keeping a `Style` alive in some shared `Rc` or dropping it later will cause this. `Style -> inner.manager: StyleManager -> inner: StyleManagerBuilder`. 

Since `StyleManager` is clone and keeps an `Rc<StyleManagerBuilder>` anyway, one could just read the data directly, without the need of a channel, from a `fn read_static_markup(&self)` on the `StyleManager`? Could look like

```rust
            let manager = StyleManager::builder()
                .writer(writer)
                .build()
                .expect("failed to create style manager.");
            ServerAppProps { manager: manager.clone() }
    //
    manager.read_static_markup()
```

Reply:
The reference cycle has been broken and now Styles hold WeakStyleManager instead of StyleManager.

StyleManager is `!Send` which means it can only live in a local rendering task.
The drop behaviour is only needed if `StaticReader` and `StaticWriter` is used to send markup to a different thread.

Which is outside of the rendering task, So it will not block each other. If the rendering task finishes, it will be dropped.

Reply:
> The reference cycle has been broken and now Styles hold WeakStyleManager instead of StyleManager.

Ah good to know. I still think, being explicit about when the style data is read would be better than implicitly trusting the manager being dropped at the right time. How much would it hurt performance to put the ManagerBuilder in an `Arc` instead? I would think that atomics on wasm are compiled to normal memory accesses anyway, iirc, and don't have any impact on performance due to the assumption of single-threaded access.

Reply:
I have made it to use a Mutex instead of channels. I kept the StyleReader and StyleWriter so they use a consume-based behaviour.

This should help prevent users from sharing StyleManager between renders.